common/sgx/tcbinfo.c

```diff
@@ -327,21 +327,33 @@ static void _determine_platform_tcb_info_tcb_level(
     if (platform_tcb_level->status.AsUINT32 != OE_TCB_LEVEL_STATUS_UNKNOWN)
         return;
 
+    bool comp_svn_out_of_date = false;
+    bool pce_svn_out_of_date = false;
+
     // Compare all of the platform's comp svn values with the corresponding
     // values in the current tcb level.
     for (uint32_t i = 0; i < OE_COUNTOF(platform_tcb_level->sgx_tcb_comp_svn);
          ++i)
     {
         if (platform_tcb_level->sgx_tcb_comp_svn[i] <
             tcb_level->sgx_tcb_comp_svn[i])
-            return;
+            comp_svn_out_of_date = true;
     }
     if (platform_tcb_level->pce_svn < tcb_level->pce_svn)
+        pce_svn_out_of_date = true;
+
+    if (tcb_level->status.fields.up_to_date == 1)
+    {
+        platform_tcb_level->comp_svn_out_of_date = comp_svn_out_of_date;
+        platform_tcb_level->pce_svn_out_of_date = pce_svn_out_of_date;
```

Comment:
platform_tcb_level are specifically defined in [https://api.portal.trustedservices.intel.com/documentation#pcs-tcb-info-v2](https://api.portal.trustedservices.intel.com/documentation#pcs-tcb-info-v2) and should not be altered. I think a better approaches are:
1. For this function to return different oe_result_t from the values in comp_svn_out_of_date and pce_svn_out_of_date.
2. Add extra output parameter(s) to this function.
and then deliver the result all the way up to oe_parse_tcb_info_json() where error messages are printed.

Reply:
One possibility is to create a "verification error type" enum that would uniquely identify the reason why the SGX quote verification failed.